include/ginkgo/core/base/utils.hpp

```diff
@@ -139,6 +139,7 @@ using shared_type = std::shared_ptr<pointee<Pointer>>;
 
 
 /**
+ * @internal
```

Comment:
I think these are actually not internals, they can be used by the users instead of the standard smart pointer functions. They were designed with the users in mind, in fact.

Reply:
Yeah, that is true. But they are in the `detail` namespace.

Reply:
No, the detail namespace stops just before, in line 138.

Reply:
Yes, you are right. I got confused because I saw the return type was in the detail namespace.